lib/fastlane_core/project.rb

```diff
@@ -275,4 +276,18 @@ def parsed_info
       @parsed_info
     end
   end
+
+  # @internal to module
+  def self.xcode_list_timeout
+    (ENV['FASTLANE_XCODE_LIST_TIMEOUT'] || 10).to_i
+  end
+
+  # runs the specified command and kills it if timeouts
+  # @raises Timeout::Error if timeout is passed
+  # @returns the output
+  # FIXME: this is not working exactly as it should. gym#143
```

Comment:
The current implementation (and the new one) contain an issue. This was just a reminder of that problem. I can remove the FIXME if you want. The issue is demonstrated in the unit test were you added your "hu?" comment. I can open a new issue in github instead.